### ISSUE TITLE
ZOHO CRM integration reports "Invalid Token" sporadically

### ISSUE THREAD

**Issue by wblommaert** [solspace/craft3-freeform#1372]
### What happened?

We are using the ZOHO CRM integration on one of our forms and noticed that occasionally we receive an "INVALID_TOKEN" error in the Freeform logs.

<img width="1300" alt="Screenshot 2024-06-25 at 15 20 29" src="https://github.com/solspace/craft-freeform/assets/5678605/a6e71fb4-7119-4b62-a8ec-49b3fb2d4132">

This does not happen on every form submit, but rather from time to time. So we suspect token expirations not lining up with the way tokens are stored (if they're even stored at all for re-use?). When we check the integration in the settings, the integration itself seems properly authorized.

<img width="383" alt="Screenshot 2024-06-25 at 15 35 44" src="https://github.com/solspace/craft-freeform/assets/5678605/93d133ef-c50b-470a-95fb-fee0782850ff">

### Errors and Stack Trace (if available)

```shell
[2024-06-18T15:58:13.498983+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-19T16:37:12.576280+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-20T12:17:50.211997+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-20T13:43:06.448777+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-21T10:36:38.205324+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-21T16:13:19.035725+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-24T07:53:49.494383+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-24T10:27:25.604308+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-24T11:34:29.586862+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-24T13:50:37.068261+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-24T15:30:41.429754+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-24T21:49:58.580580+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-25T08:45:00.359574+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
[2024-06-25T12:13:45.553063+02:00] Integration.ERROR: Zoho: {"code":"INVALID_TOKEN","details":{},"message":"invalid oauth token","status":"error"}  {"integration":{"id":5,"handle":"<redacted>"}} []
```

### How can we reproduce this?

1. Install Craft CMS
2. Install Freeform Pro
3. Enable and authorize the ZOHO CRM integration using correct credentials
4. Enable it for a specific form and perform correct mapping
5. Submit the specified form
6. Occasionally the error will occur and appear in the Freeform error log

### Freeform Edition

Pro

### Freeform Version

5.3.0

### Craft Version

4.9.5

### When did this issue start?

After upgrading from older Freeform version

### Previous Freeform Version

4.x

**Comment by kjmartens** [comment]
Sorry for the trouble @wblommaert,

Would you be able to send us an email (support@solspace.com) with the `metadata` entry in your database for the Zoho integration? (`freeform_integrations` table)?

**Comment by wblommaert** [comment]
> Sorry for the trouble @wblommaert,
> 
> Would you be able to send us an email ([support@solspace.com](mailto:support@solspace.com)) with the `metadata` entry in your database for the Zoho integration? (`freeform_integrations` table)?

Hi @kjmartens, I've just sent this your way using the email address provided. Thanks!

**Comment by kjmartens** [comment]
This should now be fixed in Freeform 5.5.0. Can you please check and see if that's the case for you? 🙂
